src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -500,6 +510,325 @@ function calc_boundary_flux_by_direction_divergence!(surface_flux_values::Abstra
+                    end
+                end
+            else
+                # L2 mortars in y-direction
+                for l in eachnode(dg)
+                    for v in eachvariable(equations_parabolic)
+                        cache.mortars.u_upper[1, v, l, mortar] = flux_viscous_y[v, l,
+                                                                                nnodes(dg),
+                                                                                upper_element]
+                        cache.mortars.u_lower[1, v, l, mortar] = flux_viscous_y[v, l,
+                                                                                nnodes(dg),
+                                                                                lower_element]
+                    end
+                end
+            end
+        end
+
+        # Interpolate large element face data to small interface locations
+        if cache.mortars.large_sides[mortar] == 1 # -> large element on left side
+            leftright = 1
+            if cache.mortars.orientations[mortar] == 1
+                # L2 mortars in x-direction
+                u_large = view(flux_viscous_x, :, nnodes(dg), :, large_element)
+                element_solutions_to_mortars!(cache.mortars, mortar_l2, leftright,
+                                              mortar, u_large)
+            else
+                # L2 mortars in y-direction
+                u_large = view(flux_viscous_y, :, :, nnodes(dg), large_element)
+                element_solutions_to_mortars!(cache.mortars, mortar_l2, leftright,
+                                              mortar, u_large)
+            end
+        else # large_sides[mortar] == 2 -> large element on right side
+            leftright = 2
+            if cache.mortars.orientations[mortar] == 1
+                # L2 mortars in x-direction
+                u_large = view(flux_viscous_x, :, 1, :, large_element)
+                element_solutions_to_mortars!(cache.mortars, mortar_l2, leftright,
+                                              mortar, u_large)
+            else
+                # L2 mortars in y-direction
+                u_large = view(flux_viscous_y, :, :, 1, large_element)
+                element_solutions_to_mortars!(cache.mortars, mortar_l2, leftright,
+                                              mortar, u_large)
+            end
+        end
+    end
+
+    return nothing
+end
+
+# TODO: This is essentially not different from the implementation in standard 2D.
+# Only difference: defined `u_transformed` (equivalent of `u`) as AbstractArray.
+function prolong2mortars!(cache, u_transformed::AbstractArray,
+                          mesh::TreeMesh{2},
+                          equations_parabolic::AbstractEquationsParabolic,
+                          mortar_l2::LobattoLegendreMortarL2, surface_integral,
+                          dg::DGSEM)
+    @threaded for mortar in eachmortar(dg, cache)
```

Comment:
Do we need a different implementation for this, or could we re-use the implementation for standard 2D

https://github.com/trixi-framework/Trixi.jl/blob/a12f82da43a16d59db16063557fef245c87b6c0e/src/solvers/dgsem_tree/dg_2d.jl#L785-L788

where we would just need to specify the datatype of `u` as `AbstractArray`.

Reply:
It's basically the same implementation, isn't it? If so, it should work if you just delete this method here?

Reply:
Seems to work, tests pass.

Reply:
Resolved in [2149cfd](https://github.com/trixi-framework/Trixi.jl/pull/1571/commits/2149cfd9efd82ea086f2e4d6d138c4c683ad1c7a).